src/remote_prune

```diff
@@ -0,0 +1,198 @@
+import yaml
+import collections
+import datetime
+import os
+from dateutil.relativedelta import relativedelta
+import requests
+
+# from cosalib.aliyun import remove_aliyun_image
+# from cosalib.gcp import remove_gcp_image
+# from cosalib.prune import fetch_build_meta, delete_build
+from cosalib.s3 import S3
+from cosalib.aws import deregister_ami, delete_snapshot
+
+Build = collections.namedtuple("Build", ["id", "timestamp", "images", "arch"])
+
+def main():
+    parser = argparse.ArgumentParser(prog="coreos-assembler remote-prune")
+    parser.add_argument("--policy", default='./policy.yaml', type=str,
+                        help="Path to policy.yaml file", required=True)
+    parser.add_argument("--dry-run", dest='dry_run', help="Don't actually delete anything",
+                        action='store_true')
+    parser.add_argument("--stream", dest="stream", type=str, help="Fedora stream", required=True)
+
+    parser.add_argument("--gcp-json-key", help="GCP Service Account JSON Auth",
+                        default=os.environ.get("GCP_JSON_AUTH"))
+    parser.add_argument("--gcp-project", help="GCP Project name",
+                        default=os.environ.get("GCP_PROJECT_NAME"))
+    parser.add_argument("--aws-credentials", dest="aws_credentials", help="AWS Credentials",
+                        default=os.environ.get("AWS_CONFIG_FILE"), type=str)
+
+    # subparsers = parser.add_subparsers(dest='cmd', title='subcommands')
+    # subparsers.required = True
+
+    # s3 = subparsers.add_parser('s3', help='Prune s3 buckets')
+    # s3.add_argument("--bucket", help="Bucket name")
+    # s3.add_argument("--prefix", help="Key prefix")
+    # s3.add_argument("--force", help="Wipe s3 key ignoring the errors",
+    #                 action='store_true')
+
+    args = parser.parse_args()
+    if args.aws_credentials:
+        os.environ["AWS_CONFIG_FILE"] = args.aws_credentials
+    cloud_config = {
+        'gcp': {
+            'json-key': args.gcp_json_key,
+            'project': args.gcp_project,
+        }
+    }
+    policy = open(f"{args.policy}")
+    policy = yaml.safe_load(policy)
+    stream = args.stream
+
+    # These lists are up to date as of schema hash
+    # 4c19aed3b3d84af278780bff63728510bb3e70613e4c4eef8cabd7939eb31bd8. If changing
+    # this hash, ensure that the list of supported and unsupported artifacts below
+    # is up to date.
+    supported = ["amis"]
+    unsupported = ["base-oscontainer", "aliyun", "azure", "azurestack", "digitalocean", "exoscale", "gcp", "kubevirt", "hyperv", "ibmcloud"]
+
+    if stream in policy:
+        # If the build key is set in the policy file, then the cloud-uploads key must
+        # also be present, and the duration of cloud-uploads must be equal or shorter
+        if "build" in policy[stream].keys():
+            cloudUploadsCheck(policy[stream])
+        BASE_URL = "https://builds.coreos.fedoraproject.org/prod/streams/" + stream
+        buildJsonData = getBuildsForStream(stream)
```

Comment:
Let's not use `cosa buildfetch` for this. We don't need most of the functionality of `buildfetch` because we're operating across a wide array of builds, not just e.g. the latest. I think fetching `builds.json` directly ourselves will be cleaner.